packages/compass/src/app/index.tsx

```diff
@@ -29,6 +30,15 @@ window.addEventListener('error', (event) => {
   );
 });
 
+window.addEventListener('unhandledrejection', (event) => {
+  event.preventDefault();
+  const error = ensureError(event.reason);
+  void ipcRenderer?.call('compass:error:fatal', {
```

Comment:
This logs to the terminal like:

```
MongoDB Compass Dev Local has encountered an unexpected error: MongoDB Compass Dev Local version 0.0.1-dev.0
Stacktrace:
NonError: 'fake unhandled rejection'
    at module.exports (webpack://mongodb-compass/./node_modules/ensure-error/index.js?:20:10)
    at handleError (webpack://mongodb-compass/./src/main/handle-error.ts?:17:60)
    at process.handleUnhandledRejection (webpack://mongodb-compass/./src/main/handle-unhandled-rejection.ts?:10:67)
    at process.emit (node:events:517:28)
    at emit (node:internal/process/promises:149:20)
    at processPromiseRejections (node:internal/process/promises:294:27)
    at process.processTicksAndRejections (node:internal/process/task_queues:96:32)
```

and to the log as:

```
{"t":{"$date":"2024-02-14T14:33:29.053Z"},"s":"F","c":"COMPASS-MAIN","id":1001000002,"ctx":"app","msg":"Uncaught exception: 'fake promise rejection'","attr":{"message":"'fake promise rejection'","stack":"NonError: 'fake promise rejection'\n    at module.exports (webpack://MongoDBCompass/./node_modules/ensure-error/index.js?:20:10)\n    at eval (webpack://MongoDBCompass/./src/app/index.tsx?:66:68)"}}
```

Reply:
ie. it says it is an uncaught exception because that's what we do for compass:error:fatal: https://github.com/mongodb-js/compass/blob/e34231ebfbdb2b021341a4f948d3aae8dc68915d/packages/compass/src/main/logging.ts#L101